pkg/operator/controller/ingress_status_test.go

```diff
@@ -59,136 +68,113 @@ func provisionedLBservice(owner string) *corev1.Service {
+	return corev1.Event{
+		Type:    "Warning",
+		Reason:  "CreatingLoadBalancerFailed",
+		Message: "failed to ensure load balancer for service openshift-ingress/router-default: TooManyLoadBalancers: Exceeded quotaof account",
+		Source: corev1.EventSource{
+			Component: "service-controller",
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind: "Service",
+			Name: service,
+		},
 	}
 }
 
-func indexerContains(indexer toolscache.Indexer, name, value string) bool {
-	objects, err := indexer.ByIndex(name, value)
-	if err != nil {
-		return false
+func schedulerEvent() corev1.Event {
+	return corev1.Event{
+		Type:   "Normal",
+		Reason: "Scheduled",
+		Source: corev1.EventSource{
+			Component: "default-scheduler",
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind: "Pod",
+			Name: "router-default-1",
+		},
 	}
-	return len(objects) > 0
 }
 
-func makeIndexers(funcs map[string]client.IndexerFunc) toolscache.Indexers {
-	indexers := map[string]toolscache.IndexFunc{}
-	for name := range funcs {
-		fn := funcs[name]
-		indexers[name] = func(obj interface{}) ([]string, error) {
-			return fn(obj.(runtime.Object)), nil
-		}
+func cond(t string, status operatorv1.ConditionStatus, reason string) operatorv1.OperatorCondition {
+	return operatorv1.OperatorCondition{
+		Type:   t,
+		Status: status,
+		Reason: reason,
 	}
-	return indexers
 }
 
 func TestComputeLoadBalancerStatus(t *testing.T) {
 	tests := []struct {
-		desc       string
-		controller string
-		strat      operatorv1.EndpointPublishingStrategyType
-		services   []*corev1.Service
+		name       string
+		controller *operatorv1.IngressController
+		service    *corev1.Service
+		events     []corev1.Event
 		expect     []operatorv1.OperatorCondition
 	}{
 		{
-			desc:       "provisioned/ready",
-			controller: "default",
-			strat:      operatorv1.LoadBalancerServiceStrategyType,
-			services: []*corev1.Service{
-				provisionedLBservice("secondary"),
-				provisionedLBservice("default"),
-				clusterIPservice("default"),
-			},
+			name:       "lb provisioned",
+			controller: ingressController("default", operatorv1.LoadBalancerServiceStrategyType),
+			service:    provisionedLBservice("default"),
 			expect: []operatorv1.OperatorCondition{
-				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "HasLoadBalancerEndpointPublishingStrategy"),
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "WantedByEndpointPublishingStrategy"),
 				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionTrue, "LoadBalancerProvisioned"),
 			},
 		},
 		{
-			desc:       "pending/unready",
-			controller: "default",
-			strat:      operatorv1.LoadBalancerServiceStrategyType,
-			services: []*corev1.Service{
-				provisionedLBservice("secondary"),
-				pendingLBService("default"),
-				clusterIPservice("default"),
-			},
+			name:       "lb pending, no events",
```

Comment:
Can this be "no events for current lb" with `events: []corev1.Event{schedulerEvent(), failedCreateLBEvent("secondary")}`?

Reply:
Great improvement, fixed